includes/class-wc-payments-account.php

```diff
@@ -1048,6 +1048,15 @@ public function maybe_handle_onboarding() {
 				return;
 			}
 
+			if ( isset( $_GET['wcpay-reset-account'] ) ) {
+				$test_mode = WC_Payments_Onboarding_Service::is_test_mode_enabled();
```

Comment:
I wonder what happens if test_mode is enabled? for dev mode we don't allow it, but for live account in test_mode we do allow deletion, right?

Reply:
I don't think it matters for a live account (because of the way the server logic is implemented, test accounts cannot act in live mode, but live accounts can act in either test or live mode).

I also think it's not possible for a live account to access this when in test mode, because the Account Tools only display when the account is not fully onboarded, and the user cannot access the settings to modify the Test Mode value if not fully onboarded.

The server also has some security in place against pretending a live account is actually a test one.